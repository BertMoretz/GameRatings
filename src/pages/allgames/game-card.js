import React from 'react';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import axios from 'axios'
import GridListTile from '@material-ui/core/GridListTile';
import GridListTileBar from '@material-ui/core/GridListTileBar';
import IconButton from '@material-ui/core/IconButton';
import Link from '@material-ui/core/Link';

import style from './all-games.css'


export const GameCard = ({ game, handleDetailsClick }) => (
  <GridListTile key={game.name} className={style.elem}>
      <a href="#" className={style.coverlink} onClick={handleDetailsClick}>
        <div className={style.detailscover}>
            Details
        </div>

        <img src={game.cover? "https://images.igdb.com/igdb/image/upload/t_cover_big/" + game.cover.image_id + ".jpg" : ""} alt={game.name} className={style.elem} />


      <GridListTileBar
        className={style.caption}
        title={game.name}
        subtitle={<span> {game.genres? game.genres[0].name: "  "}</span>}
        actionIcon={
          <IconButton >
            <div className={style.score}>
            { ('aggregated_rating' in game)? parseInt(game.aggregated_rating )+"%" : "tbd" }
            </div>
          </IconButton>
        }
      />
     </a>
  </GridListTile>
)
