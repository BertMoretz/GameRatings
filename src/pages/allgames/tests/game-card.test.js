import React from 'react'
import { GameCard } from '../game-card'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


import { shallow, mount, render } from 'enzyme';

test('Grid renders as container', () => {

    const shallowComponent = shallow(<GameCard game={{origin: ''}} handleDetailsClick={() =>{}} />)
    const mountComponent = mount(<GameCard game={{origin: ''}} handleDetailsClick={() =>{}} />)
    expect(shallowComponent).toBeDefined()
    expect(shallowComponent.debug()).toMatchSnapshot()

	    // Test lifecycle method calls and all possible component states
	    // console.log(shallowComponent.debug())
	    // console.log(mountComponent.debug())
})
