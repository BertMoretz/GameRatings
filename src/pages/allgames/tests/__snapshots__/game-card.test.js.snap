// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
"<WithStyles(GridListTile) className=\\"elem\\">
  <a href=\\"#\\" className=\\"coverlink\\" onClick={[Function: handleDetailsClick]}>
    <div className=\\"detailscover\\">
      Details
    </div>
    <img src=\\"\\" alt={[undefined]} className=\\"elem\\" />
    <WithStyles(GridListTileBar) className=\\"caption\\" title={[undefined]} subtitle={{...}} actionIcon={{...}} />
  </a>
</WithStyles(GridListTile)>"
`;
